addons/product/static/src/js/pricelist_report/product_pricelist_report.js

```diff
@@ -0,0 +1,213 @@
+                this.MAX_QTY
+            );
+            await this.action.doAction(customNotification("warning", message));
+            return;
+        }
+
+        const qty = parseInt($("input.add-quantity-input")[0].value);
+        if (qty && qty > 0) {
+            // Check qty already exist.
+            if (this.quantities.indexOf(qty) === -1) {
+                this.quantities.push(qty);
+                this.quantities = this.quantities.sort((a, b) => a - b);
+                this.renderHtml();
+            } else {
+                let message = sprintf(this.env._t("Quantity already present (%s)."), qty);
+                await this.action.doAction(customNotification("info", message));
+            }
+        } else {
+            await this.action.doAction(
+                customNotification("info", this.env._t("Please enter a positive whole number."))
+            );
+        }
+    }
+
+    onClickLink(ev) {
+        ev.preventDefault();
+
+        let classes = ev.target.getAttribute("class", "");
+        let resModel = ev.target.getAttribute("data-model", "");
+        let resId = ev.target.getAttribute("data-res-id", "");
+
+        if (classes && classes.includes("o_action") && resModel && resId) {
+            this.action.doAction({
+                type: 'ir.actions.act_window',
+                res_model: resModel,
+                res_id: parseInt(resId),
+                views: [[false, 'form']],
+                target: 'self',
+            });
+        }
+    }
+
+    onClickPrint() {
+        this.action.doAction({
+            type: 'ir.actions.report',
+            report_type: 'qweb-pdf',
+            report_name: 'product.report_pricelist',
+            report_file: 'product.report_pricelist',
+            data: this.reportParams,
+        });
+    }
+
+    async onClickRemoveQty(ev) {
+        if (this.quantities.length <= 1) {
+            await this.action.doAction(
+                customNotification("warning", this.env._t("You must leave at least one quantity."))
+            );
+            return;
+        }
+
+        const qty = parseInt(ev.srcElement.parentElement.childNodes[0].data);
+        this.quantities = this.quantities.filter(q => q !== qty);
+        this.renderHtml();
+    }
+
+    onSelectPricelist(ev) {
+        this.state.selectedPricelist = this.pricelists.filter(pricelist => 
+            pricelist.id === parseInt(ev.target.value)
+        )[0];
+
+        this.renderHtml();
+    }
+
+    onToggleDisplayPricelist() {
+        this.state.displayPricelistTitle = !this.displayPricelistTitle;
+        this.renderHtml();
+    }
+}
+
+ProductPricelistReport.props = {
+    action: { type: Object },
+    actionId: { type: Number, optional: true },
+    className: { type: String },
+    state: { type: Object, optional: true },
+    globalState: { type: Object, optional: true },
+}
```

Comment:
I don't think all of this is needed here :eyes: 
```suggestion
ProductPricelistReport.props = {
    action: { type: Object },
    "*": true,
}
```

Reply:
with the `"*": true` the warning is gone ! :)